sv2/channels-sv2/src/server/jobs/job_store.rs

```diff
@@ -47,72 +66,57 @@ pub trait JobStore<T: Job>: Send + Sync + Debug {
     fn get_future_job_id_from_template_id(&self, template_id: u64) -> Option<u32>;
 
     /// Returns an owned copy of the currently active job, if any.
-    fn get_active_job(&self) -> Option<T>;
+    fn get_job(&self, job_id: u32) -> JobLifecycleState<JobOut>;
 
     /// Returns true if there are any future jobs, false otherwise.
     fn has_future_jobs(&self) -> bool;
 
-    /// Returns an owned copy of a future job from its job ID, if any.
-    fn get_future_job(&self, job_id: u32) -> Option<T>;
-
-    /// Returns true if there are any past jobs, false otherwise.
-    fn has_past_jobs(&self) -> bool;
-
-    /// Returns an owned copy of a past job from its job ID, if any.
-    fn get_past_job(&self, job_id: u32) -> Option<T>;
+    /// Removes an owned copy of a future job from its job ID, if any.
+    fn remove_future_job(&mut self, job_id: u32) -> Option<EitherJob<'a>>;
 
-    /// Returns true if there are any stale jobs, false otherwise.
-    fn has_stale_jobs(&self) -> bool;
-
-    /// Returns an owned copy of a stale job from its job ID, if any.
-    fn get_stale_job(&self, job_id: u32) -> Option<T>;
+    fn try_validate_job(
```

Comment:
I went this route because the current trait kinda nudges you toward having the same structure as DefaultJobStore. In it's most basic form I beleive the JobStore should allow you to:
1. add an active job
2. add and activate a future job
3. make jobs stale
4. retrieve jobs for validation

The extra methods are bloat imo. 
In our case we use timestamp based job ids, and it could be possible---albeit a very low chance---that we have duplicate job_ids. Also, assuming there is access to the underlying JobStore, the user could override share validation to allow for merged mining